### ISSUE TITLE
Function 'git_repo_exists' returns False even though the repository exits

### ISSUE THREAD

**Issue by teojgo** [eth-cscs/reframe#191]
The problem seems to happen for remote repositories that have many branches

**Comment by vkarak** [comment]
Can you add here the exact test case that triggers this problem?

**Comment by teojgo** [comment]
```
import reframe.utility.os as os_ext
os_ext.git_repo_exists('https://github.com/tensorflow/models.git')
```

**Comment by teojgo** [comment]
More specifically when debugging the actual command a `subprocess.TimeoutExpired` exception is thrown. This must happen because the `subprocess.Popen` is called with `stdout=subprocess.PIPE`
 and `stderr=subprocess.PIPE` and according to the Python documentation if the output is large, the subprocess deadlocks.

**Comment by vkarak** [comment]
How large is the output the command we execute? There could be a "quiet" option that we could use perhaps.

**Comment by teojgo** [comment]
It is quite large since the `git ls-remote` sees all the branches at the remote repo. The quiet option just does not print the `==STDERR==` label and possibly also the stderr if it exists. For me using `git ls-remote -h` worked, because `-h` lists only the ref/heads and thus the output is much smaller.

**Comment by vkarak** [comment]
Could you submit a patch for this?

**Comment by teojgo** [comment]
Fixed by #192
